From c2b075897026f7e720a4650f9826a1b81f4c3e22 Mon Sep 17 00:00:00 2001
From: Javier Pena <jpena@redhat.com>
Date: Wed, 20 May 2015 11:32:18 +0200
Subject: [PATCH] Fix xmldsig import

Following pysaml2 commit 9af3252035484f4a8c624eba0f35b68280d43fd2 ,
xmldsig was moved under saml2, no longer being at the top level.
Craeting a conditional import, so it can work now and when the
pysaml2 commit is included in a new release.

Change-Id: If01a93a18cef6159adad50496e6335e88f86024d
---
 keystone/contrib/federation/idp.py        | 5 ++++-
 keystone/tests/unit/test_v3_federation.py | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/keystone/contrib/federation/idp.py b/keystone/contrib/federation/idp.py
index ef4b2e9..dedb525 100644
--- a/keystone/contrib/federation/idp.py
+++ b/keystone/contrib/federation/idp.py
@@ -18,6 +18,7 @@ import uuid
 from oslo_config import cfg
 from oslo_log import log
 from oslo_utils import timeutils
+from oslo_utils import importutils
 import saml2
 from saml2 import client_base
 from saml2 import md
@@ -26,7 +27,9 @@ from saml2 import saml
 from saml2 import samlp
 from saml2.schema import soapenv
 from saml2 import sigver
-import xmldsig
+xmldsig = importutils.try_import("xmldsig")
+if not xmldsig:
+    xmldsig = importutils.try_import("saml2.xmldsig")
 
 from keystone import exception
 from keystone.i18n import _, _LE
diff --git a/keystone/tests/unit/test_v3_federation.py b/keystone/tests/unit/test_v3_federation.py
index a8a258b..22071a8 100644
--- a/keystone/tests/unit/test_v3_federation.py
+++ b/keystone/tests/unit/test_v3_federation.py
@@ -26,7 +26,10 @@ import saml2
 from saml2 import saml
 from saml2 import sigver
 from six.moves import range, urllib, zip
-import xmldsig
+from oslo_utils import importutils
+xmldsig = importutils.try_import("xmldsig")
+if not xmldsig:
+    xmldsig = importutils.try_import("saml2.xmldsig")
 
 from keystone.auth import controllers as auth_controllers
 from keystone.auth.plugins import mapped
-- 
2.1.0

